### PR TITLE
Models for database 

### LINKED ISSUE #105: Optimizing the Database  (user and project) model
### 2. We need to create a user model with it and set up an optional relationship with the project model (database relations will be one - too many ). 

code

### PR DESCRIPTION
**Pull request by leewan-09**
### Fixes Issue
fixes #105 

- Created Models for database 
- Changed `prisma.js` to typescript and improvements

### CODE REVIEW

File: pages/api/db/prisma.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { PrismaClient } from '@prisma/client';
```
Comment (rohitdasu) [inline pages/api/db/prisma.ts:1]: @leewan-09 i was curious if we could move this file to /lib
any special reason for keeping it inside api/ folder?

Comment (AdityaPainuli) [reply pages/api/db/prisma.ts:1]: we can sure put it outside API folder I think there would not be any problem with that.

Comment (rohitdasu) [reply pages/api/db/prisma.ts:1]: i think /lib folder would be best for it

Comment (leewan-09) [reply pages/api/db/prisma.ts:1]: We could move but it is our DB connection file I would like to keep it like that.

Comment (AdityaPainuli) [reply pages/api/db/prisma.ts:1]: yup, we can go on with it @leewan-09 try to put it on /lib and insure everything is working fine as expected .

Comment (AdityaPainuli) [reply pages/api/db/prisma.ts:1]: > We could move but it is our DB connection file I would like to keep it like that.

yes I don't think it's a good idea to put it inside API folder

Comment (leewan-09) [reply pages/api/db/prisma.ts:1]: From [here](https://www.youtube.com/c/TheoBrowne1017) i learnt that...

Comment (AdityaPainuli) [reply pages/api/db/prisma.ts:1]: yup, it's just a naming convention everyone can have different opinions on it. My main reason to say this is that the API folder is going to have too many routes and we already have an /lib folder I guess so it's no harm putting it that way.

**Comment by AdityaPainuli** [comment]
I think we can merge it as the schema model is looking good right now. we will wait till one more review then merge it great job @leewan-09 🚀🚀

**Review by rohitdasu — APPROVED** [review]
Looks good for me too 👍

**Pull request merged**